### ISSUE TITLE
hlint customization confusions, HLint.hs error reporting

### ISSUE THREAD

**Issue by simonmichael** [ndmitchell/hlint#256]
Trying to customise hlint, I've run into some confusion and questions.
https://github.com/ndmitchell/hlint#customizing-the-hints gives a brief example of a Hlint.hs file:

``` hs
import "hint" HLint.Default
import "hint" HLint.Dollar
```

Should that package name be "hlint" rather than "hint" ?
Is the file name case sensitive ? `hlint.hs` also works here, but perhaps that's because I'm testing on a mac.
In my current project (a stack project with hlint installed), the package name qualifier is required, I'm not sure why.
More error reporting about HLint.hs issues would be useful. Eg anything following the `import` keyword is accepted without error, even with `-v`.
Is HLint.hs actually a haskell file, or a custom config syntax ?

Thank you very much for hlint!

**Comment by simonmichael** [comment]
More (please excuse the laundry list): 

it seems both "hint" and "hlint" work after `import`, but not anything else.

Without the HLint.hs file, I see some hints. With a HLint.hs file importing HLint.Default, I expected to see the same hints, but I see none. If I import HLint.Dollar, I do see the dollar hints. It would be good to know which hints hlint is currently using. Also a --list command, or an easy way to see the data directory path, would be useful to see which hints are available.

**Comment by simonmichael** [comment]
It seems hlint uses data-dir/HLint.hs by default. Adding `import "hint" HLint.Builtin.All` to my HLint.hs reproduces the default warnings. I'm wondering whether importing HLint.Default adds anything to this.

**Comment by ndmitchell** [comment]
I originally said you had to import `"hint"`. Later on I realised that typing that as `"hlint"` is almost inevitable, given the concepts going around people's head at that point. So I accept both.

Any import which doesn't have `"hint"` or `"hlint"` in it is considered to be a declaration of scoping, so if you do `import Data.List` you are saying that some of the identifiers in this file might come from `Data.List`, and you aren't going to qualify them.

There are "builtin" hints, e.g. eta reduction, which are wired into HLint. Those can be obtained through `HLint.Builtin.All`. There are also hints defined in terms of the `==>` language, e.g. `map f . map g ==> map (f . g)`., those are available from `HLint.Default`, and won't be in scope if you don't import them.

Certainly the config file declaration format is a bit of a mess. I have never really got to grips with what its semantics should be or who should be extending it - quite a few of the open tickets are around this area.

**Comment by ndmitchell** [comment]
I entirely rewrote the configuration mechanism with `.hlint.yaml` files. By no longer abusing Haskell syntax hopefully all the confusing bits have gone away. If not, please do reopen this ticket with more suggestions.
